src/physics/base/ParticleTrackView.hh

```diff
@@ -0,0 +1,87 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file ParticleTrackView.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "ParticleStatePointers.hh"
+#include "ParticleParamsPointers.hh"
+#include "base/Macros.hh"
+#include "base/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Read/write view to the physical properties of a single particle track.
+ *
+ * These functions should be used in each physics Process.
+ */
+class ParticleTrackView
+{
+  public:
+    //@{
+    //! Type aliases
+    using Initializer_t = ParticleTrackState;
+    //@}
+
+  public:
+    // Construct from "dynamic" state and "static" particle definitions
+    inline CELER_FUNCTION
+    ParticleTrackView(const ParticleParamsPointers& params,
+                      const ParticleStatePointers&  states,
```

Comment:
Slightly (minorly?) confusing nomenclature.  ParticleStatePointers is an accessor to a set of ParticleState while ParticleParamsPointers is an accessor to a set of ParticleDef (rather than a set of ParticleParam[s]).  Is the Param/Def distinction intentional?

Reply:
My thinking is that `Params` is a single object that has-a vector of `Def`s. The fact that `ParticleStatePointers` has a vector of `ParticleTrackState` objects is an implementation detail; we could conceivably move to a "struct of arrays" and keep the same `ParticleStatePointers`/`ParticleTrackView` interface.

Mainly, until the paradigm breaks down, I'm just trying to keep the "Params" vs "State" nomenclature to reflect the "shared" vs "track-specific" data sets.  https://github.com/celeritas-project/celeritas/wiki/Development#data-management :

Reply:
I think confusion is that eventhough the pattern is the same `Particle....Pointers` the cardinality is completely different, i.e.
```
ParticleParamsPointers -> one value (Params) for **all** tracks
ParticleStatePointers -> one value (State) for **each** tracks
```
(or am I confused even further).

Reply:
Closing after our zoom discussion.